particulate_matter.py

```diff
@@ -0,0 +1,196 @@
+                           update_interval=config.get(CONF_SCAN_INTERVAL))
+
+    dev = []
+
+    if config.get(CONF_OFFSET_PM1) > 0:
+        dev.append(ParticulateMatterSensor(coll, "PM1.0",
+                                           config.get(CONF_OFFSET_PM1)))
+    if config.get(CONF_OFFSET_PM2_5) > 0:
+        dev.append(ParticulateMatterSensor(coll, "PM2.5",
+                                           config.get(CONF_OFFSET_PM2_5)))
+    if config.get(CONF_OFFSET_PM10) > 0:
+        dev.append(ParticulateMatterSensor(coll, "PM10",
+                                           config.get(CONF_OFFSET_PM10)))
+
+    add_devices(dev)
+
+
+class ParticulateMatterSensor(Entity):
+    """Representation of an Particulate matter sensor."""
+
+    # pylint: disable=too-many-arguments
+
+    def __init__(self, pmDataCollector, name, offset):
+        """Initialize a new PM sensor."""
+        self._name = name
+        self._state = 0
+        pmDataCollector.add_listener(offset, self.update_data)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return "µg/m³"
+
+    def update_data(self, value):
+        """Update the date to the given value.
+
+        This is used from the PMDataCollector background thread
+        """
+        self._state = value
+        if not self.should_poll:
+            self.update_ha_state()
+        _LOGGER.debug("State of %s updated to %s", self._name, self._state)
+
+
+class PMDataCollector():
+    """Controls the serial interface and reads data from teh sensor."""
+
+# pylint: disable=too-many-arguments
+
+    def __init__(self,
+                 serialdevice,
+                 start_sequence,
+                 record_length,
+                 dtr=False,
+                 start_delay=10,
+                 baud_rate=9600,
+                 update_interval=300):
+        """Initialize the data collector based on the given parameters."""
+        import serial
+
+        self.record_length = record_length
+        self.start_sequence = bytes([0x32, 0x3d, 0x00, 0x1c])
+        self.update_interval = update_interval
+        self.listeners = []
+        self.dtr = dtr
+        self.start_delay = start_delay
+
+        self.ser = serial.Serial(port=serialdevice,
+                                 baudrate=baud_rate,
+                                 parity=serial.PARITY_NONE,
+                                 stopbits=serial.STOPBITS_ONE,
+                                 bytesize=serial.EIGHTBITS)
+
+        # Update date in using a background thread
+        thread = threading.Thread(target=self.refresh, args=())
+        thread.daemon = True
+        thread.start()
+
+    def refresh(self):
```

Comment:
Letting each sensor call the update method is a bit problematic as they would read different data frames.  This would result in inconsistent states (e.g. PM10 could be lower than PM2.5 when read from different data blocks). What is the recommended way to synchronise updates on multiple sensors?